subscription-service/src/main/kotlin/com/egm/stellio/subscription/service/NotificationService.kt

```diff
@@ -70,32 +73,50 @@ class NotificationService(
         )
         val uri = subscription.notification.endpoint.uri.toString()
         logger.info("Notification is about to be sent to $uri for subscription ${subscription.id}")
-        val request =
-            WebClient.create(uri).post().contentType(mediaType).headers {
-                if (mediaType == MediaType.APPLICATION_JSON) {
-                    it.set(HttpHeaders.LINK, subscriptionService.getContextsLink(subscription))
-                }
-                if (tenantName != DEFAULT_TENANT_NAME)
-                    it.set(NGSILD_TENANT_HEADER, tenantName)
-                subscription.notification.endpoint.receiverInfo?.forEach { endpointInfo ->
-                    it.set(endpointInfo.key, endpointInfo.value)
+
+        val headerMap: MutableMap<String, String> = emptyMap<String, String>().toMutableMap()
+        if (mediaType == MediaType.APPLICATION_JSON) {
+            headerMap[HttpHeaders.LINK] = subscriptionService.getContextsLink(subscription)
+        }
+        if (tenantName != DEFAULT_TENANT_NAME)
+            headerMap[NGSILD_TENANT_HEADER] = tenantName
+        subscription.notification.endpoint.receiverInfo?.forEach { endpointInfo ->
+            headerMap[endpointInfo.key] = endpointInfo.value
+        }
+
+        val result =
+            kotlin.runCatching {
+                if (uri.startsWith(Mqtt.SCHEME.MQTT)) {
+                    headerMap["Content-Type"] = mediaType.toString() // could be common with line 99 ?
+                    Triple(
+                        subscription,
+                        notification,
+                        mqttNotificationService.mqttNotifier(
+                            notification = notification,
+                            subscription = subscription,
+                            headers = headerMap
+                        )
+                    )
+                } else {
+                    val request =
```

Comment:
a good opportunity to add an `HttpNotificationService`?

Reply:
Ca serait intéressant surtout pour séparer les tests qui check http de ceux de la logique du service.
Mais ça m'as l'air assez lourd comme changement.